jobs/selfcare-onboarding-consumer/src/services/processor.ts

```diff
@@ -0,0 +1,61 @@
+import { KafkaMessage } from "kafkajs";
+import { EventPayload } from "../model/institution-event.js";
+import { RefreshableInteropToken, ORIGIN_IPA } from "@interop-be-reports/commons";
+import { TenantProcessService } from "./tenantProcessService.js";
+import { InteropContext } from "../model/interop-context.js";
+import { v4 as uuidv4 } from "uuid";
+import { SelfcareTenantSeed } from "../model/tenant-process.js";
+import { error, info, warn } from "../utils/logger.js";
+
+export const processMessage = (refreshableToken: RefreshableInteropToken, tenantProcess: TenantProcessService, productName: string) => async (message: KafkaMessage, partition: number): Promise<void> => {
+  const correlationId = uuidv4()
+
+  try {
+
+    info(correlationId, `Consuming message for partition ${partition} with offset ${message.offset}`)
+
+    if (!message.value) {
+      warn(correlationId, `Empty message for partition ${partition} with offset ${message.offset}`)
+      return
+    }
+
+    const stringPayload = message.value.toString()
+    const jsonPayload = JSON.parse(stringPayload);
+
+    // Process only messages of our product
+    // Note: Filtered before parsing to avoid errors on an unexpected messages that we are not interested in 
+    if (jsonPayload.product !== productName) {
+      info(correlationId, `Skipping message for partition ${partition} with offset ${message.offset} - Not required product: ${jsonPayload.product}`)
+      return
+    }
+
+    const parsed = EventPayload.safeParse(jsonPayload);
+
+    if (parsed.success) {
+      const token = await refreshableToken.get()
+      const context: InteropContext = { bearerToken: token.serialized, correlationId: uuidv4() }
+
+      const institution = parsed.data.institution
+
+      const seed: SelfcareTenantSeed = {
+        externalId: {
+          origin: institution.origin,
+          value: institution.origin == ORIGIN_IPA ? institution.subUnitCode || institution.originId : institution.taxCode
+        },
+        selfcareId: parsed.data.internalIstitutionID,
+        name: institution.description
+      }
+      await tenantProcess.selfcareUpsertTenant(seed, context)
```

Comment:
Why aren't dates passed through as well (e.g. `createdAt`)?

Reply:
The `createdAt` field in Interop indicates when the tenant has been created in our platform, while the one in the message refers to the creation of the contract in Selfcare.

I just covered known use cases for now, if you think more details may be useful we can integrate them in future activities 👍 
(I would not include them now because they require several reworks on persistence modelling and existing endpoints and I would prefer to close one requirement at a time)

Actually, it's a good idea to remove the unused fields from message modelling

Reply:
To be able to process tenants for the metrics requested by the DtD, we need to know the Self Care onboarding date. I agree to avoid bloating this PR with more stuff, but if this use case is not covered, we should open a new issue to manage that before working on the metrics